src-tauri/src/main.rs

```diff
@@ -171,6 +181,11 @@ fn simple_command_with_result(argument: String, state: State<NanaState>) -> Resu
 
     let result = result.map(|x| x.into_value(Span { start: 0, end: 0 }));
 
+    if let Ok(result) = &result {
+        let mut card_cache = state.card_cache.lock();
+        card_cache.insert(card_id, result.clone());
```

Comment:
`unlock()`? Or does that happen automatically in the drop?

Reply:
Yeah it happens automatically in the drop. `lock()` returns a MutexGuard: https://doc.rust-lang.org/std/sync/struct.MutexGuard.html